camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/mapred/EtlInputFormat.java

```diff
@@ -461,6 +464,17 @@ private void writeRequests(List<CamusRequest> requests, JobContext context) thro
     }
     return offsetKeysMap;
   }
+  
+  private void refreshPartitionMetadata(PartitionMetadata partitionMetadata, TopicMetadata topicMetadata, JobContext context) {
+    List<TopicMetadata> topicMetadataList = this.getKafkaMetadata(context, Collections.singletonList(topicMetadata.topic()));
+    topicMetadata = topicMetadataList.get(0);
+    for(PartitionMetadata metadataPerPartition : topicMetadata.partitionsMetadata()) {
+      if (metadataPerPartition.partitionId() == partitionMetadata.partitionId()) {
+        partitionMetadata = metadataPerPartition;
```

Comment:
This has no effect to partitionMetadata in line 265, right? java is pass by value.